src/utilities/EventEmitter.test.ts

```diff
@@ -0,0 +1,242 @@
+    });
+
+    it('Returns all listeners for a given event', () => {
+      expect(eventEmitter.listeners('myEvent')).toHaveLength(3);
+      expect(eventEmitter.listeners('myOtherEvent')).toBe(null);
+      eventEmitter.on('myEvent', listener);
+      expect(eventEmitter.listeners('myEvent')).toHaveLength(4);
+      eventEmitter.once('myEvent', listener);
+      expect(eventEmitter.listeners('myEvent')).toHaveLength(5);
+    });
+
+  });
+
+  describe('calling the once method', () => {
+    beforeAll(() => {
+      eventEmitter = new EventEmitter();
+    });
+
+    it('Adds a listener to anyOnce on calling `once` with a listener', () => {
+      eventEmitter.once(listener);
+      expect(eventEmitter.anyOnce).toHaveLength(1);
+    });
+
+    it('Adds a listener to an eventOnce on calling `once` with a listener and event name', () => {
+      eventEmitter.once('myEvent', listener);
+      expect(eventEmitter.eventsOnce['myEvent']).toHaveLength(1);
+    });
+
+    it('Adds a listener to multiple eventOnce fields on calling `once` with a listener and event name; And after emitting any of the events, all are removed', () => {
+      eventEmitter.once(['myEvent', 'myOtherEvent', 'myThirdEvent'], listener);
+      expect(eventEmitter.eventsOnce['myEvent']).toHaveLength(1);
+      expect(eventEmitter.eventsOnce['myOtherEvent']).toHaveLength(1);
+      expect(eventEmitter.eventsOnce['myThirdEvent']).toHaveLength(1);
+      expect(eventEmitter.emit('myEvent'));
+      expect(eventEmitter.eventsOnce['myEvent']).toBe(undefined);
+      expect(eventEmitter.eventsOnce['myOtherEvent']).toBe(undefined);
+      expect(eventEmitter.eventsOnce['myThirdEvent']).toBe(undefined);
+    });
+  });
+
+  describe('calling the emit method', () => {
+    beforeEach(() => {
+      eventEmitter = new EventEmitter();
+      eventEmitter.once(listener);
+    });
+
+    it('Emits an event on being called', () => {
+      let target: boolean | undefined;
+      const targetMutatorListener = () => target = true;
+      eventEmitter.on('myEvent', targetMutatorListener);
+      expect(eventEmitter.listeners('myEvent')).toContain(targetMutatorListener);
+      expect(target).toBe(undefined);
+      eventEmitter.emit('myEvent');
+      expect(target).toBe(true);
+      // anyOnce must also be emptied
+      expect(eventEmitter.anyOnce).toStrictEqual([]);
+    });
+
+    it('Emits an event and removes it on being called for a once operation', () => {
+      let target: boolean | undefined;
+      const targetMutatorListener = () => target = true;
+      eventEmitter.once('myEvent', targetMutatorListener);
+      expect(eventEmitter.listeners('myEvent')).toContain(targetMutatorListener);
+      expect(target).toBe(undefined);
+      eventEmitter.emit('myEvent');
+      expect(target).toBe(true);
+      expect(eventEmitter.listeners('myEvent')).not.toContain(targetMutatorListener);
+      expect(eventEmitter.anyOnce).toStrictEqual([]);
+    });
+  });
+
+
+
+});
```

Comment:
Some spacing issues in this file, is this not checked by prettier? (there is some missing newlines between `it` calls and there is this big gap here.

Reply:
It doesn't seem to be checked by prettier

Reply:
I see, perhaps prettier does not check newlines. either way, it would be great to make the spacing more consistent in the file – newline after `it` always and no big breaks like above.